File: test_plugins/python_test_package/python_test_package/drill_location_listener.py
#!/usr/bin/env python3

import rclpy
from geometry_msgs.msg import PointStamped
from rclpy.node import Node


class DrillLocationListener(Node):
    def __init__(self) -> None:
        super().__init__("drill_location_listener")
        self._subscription = self.create_subscription(
            PointStamped,
            "/ambf/env/plugin/volumetric_drilling/drill_location_in_volume",
            self._on_point_received,
            10,
        )
        self.get_logger().info(
            "Listening on /ambf/env/plugin/volumetric_drilling/drill_location_in_volume"
        )

    def _on_point_received(self, msg: PointStamped) -> None:
        stamp = msg.header.stamp
        self.get_logger().info(
            f"PointStamped t={stamp.sec}.{stamp.nanosec:09d} "
            f"x={msg.point.x:.6f}, y={msg.point.y:.6f}, z={msg.point.z:.6f}"
        )


def main(args=None) -> None:
    rclpy.init(args=args)
    node = DrillLocationListener()
    try:
        rclpy.spin(node)
    except KeyboardInterrupt:
        pass
    finally:
        node.destroy_node()
        rclpy.shutdown()


if __name__ == "__main__":
    main()

File: test_plugins/cpp_test_package/src/drill_location_listener.cpp
#include <functional>
#include <memory>

#include "geometry_msgs/msg/point_stamped.hpp"
#include "rclcpp/rclcpp.hpp"

class DrillLocationListener : public rclcpp::Node {
public:
  DrillLocationListener() : Node("drill_location_listener") {
    subscription_ = this->create_subscription<geometry_msgs::msg::PointStamped>(
      "/ambf/env/plugin/volumetric_drilling/drill_location_in_volume",
      10,
      std::bind(&DrillLocationListener::on_point_received, this, std::placeholders::_1));

    RCLCPP_INFO(
      this->get_logger(),
      "Listening on /ambf/env/plugin/volumetric_drilling/drill_location_in_volume");
  }

private:
  void on_point_received(const geometry_msgs::msg::PointStamped::SharedPtr msg) const {
    const auto & stamp = msg->header.stamp;
    RCLCPP_INFO(
      this->get_logger(),
      "PointStamped t=%d.%09u x=%.6f, y=%.6f, z=%.6f",
      stamp.sec,
      stamp.nanosec,
      msg->point.x,
      msg->point.y,
      msg->point.z);
  }

  rclcpp::Subscription<geometry_msgs::msg::PointStamped>::SharedPtr subscription_;
};

int main(int argc, char * argv[]) {
  rclcpp::init(argc, argv);
  rclcpp::spin(std::make_shared<DrillLocationListener>());
  rclcpp::shutdown();
  return 0;
}
